qlib/__init__.py

```diff
@@ -48,6 +48,9 @@ def init(default_conf="client", **kwargs):
     if clear_mem_cache:
         H.clear()
     C.set(default_conf, **kwargs)
+    # some loggers are created as class variable, which are initialized before config gets updated.
+    # so use `set_global_logger_level` here to update their levels.
+    set_global_logger_level(C.logging_level)
```

Comment:
Hi, I think your intention is setting the logging level of the **Loggers**(because C.logging_level is the default level of Qlib's Loggers).

But `set_global_logger_level` tries to set the logging level of **handlers**.

The Python logging mechanism is a little more complicated than user's expectation. 
Here is some [my short notes](https://github.com/you-n-g/cheatsheets/blob/13587580857acac48878b953a0f23e8e744bb1a5/language/python/log_snippet.py#L19) (boldly presuming that you can read Chinese...).  Hope it will be helpful.

If need help, you can upload a test that the previous version fails,  we can have more discussions about it.

Reply:
Hi.

Thanks for the cheetsheet, it is very helpful to understand the mechanism of python logging. And yes I can read Chinese.

My intention is just to find a way to switch off qlib log below INFO level (includes info level) from printing so that it doesn't overwhelm my own logs. By searching the doc and code I see the `logging_level' parameter can be set at `qlib.init`. However, even I set the logging level as WARN, there are still some INFO level, For example, for below code piece
```
import logging
import qlib
from qlib.contrib.data.handler import Alpha158


market = "csi300"
benchmark = "SH000300"
provider_uri = r"D:\Documents\TradeResearch\qlib_test\rqdata"


if __name__ == "__main__":
    qlib.init(provider_uri=provider_uri, region='cn', logging_level=logging.WARN)
    pre_handler = Alpha158(
        instruments=market,
        start_time="2016-01-01",
        end_time="2022-10-31",
        infer_processors=[{"class": 'Fillna', "kwargs": {"fields_group": "feature"}}],
        learn_processors=[],
    )
```

it prints
```
[19424:MainThread](2022-11-22 20:15:26,477) INFO - qlib.Initialization - [config.py:416] - default_conf: client.
[19424:MainThread](2022-11-22 20:15:27,159) INFO - qlib.Initialization - [__init__.py:74] - qlib successfully initialized based on client settings.
[19424:MainThread](2022-11-22 20:15:27,159) INFO - qlib.Initialization - [__init__.py:76] - data_path={'__DEFAULT_FREQ': WindowsPath('D:/Documents/TradeResearch/qlib_test/rqdata')}
[19424:MainThread](2022-11-22 20:15:46,901) INFO - qlib.timer - [log.py:117] - Time cost: 19.741s | Loading data Done
[19424:MainThread](2022-11-22 20:15:47,183) INFO - qlib.timer - [log.py:117] - Time cost: 0.117s | Fillna Done
[19424:MainThread](2022-11-22 20:15:47,183) INFO - qlib.timer - [log.py:117] - Time cost: 0.282s | fit & process data Done
[19424:MainThread](2022-11-22 20:15:47,184) INFO - qlib.timer - [log.py:117] - Time cost: 20.024s | Init data Done
```

Intuitively, I would expect the `logging_level` specified in `init` should take effects to all loggers.

Reply:
@qianyun210603 
I suggest adding a global config  `CONFIG_DICT`

<img width="706" alt="image" src="https://user-images.githubusercontent.com/465606/204132522-b81c4646-7d57-45fc-9372-ad68ddc3470b.png">

And then reconfig the level of all loggers in `CONFIG_DICT` at the end of initialization.

Reply:
@qianyun210603 
Besides, some tests like [this](https://monicagranbois.com/blog/python/til-how-to-unit-test-a-log-statement-in-python/) will be helpful

Reply:
> @qianyun210603 I suggest adding a global config `CONFIG_DICT`
> 
> <img alt="image" width="706" src="https://user-images.githubusercontent.com/465606/204132522-b81c4646-7d57-45fc-9372-ad68ddc3470b.png">
> 
> And then reconfig the level of all loggers in `CONFIG_DICT` at the end of initialization.

Thanks for the comment. Following this route, I would prefer to use functor to modulize to avoid too much  `global`s.
See latest version.